packages/database/src/migrations/20211006084738-AddReportsVaccineMapOverlaysCappedWS-modifies-data.js

```diff
@@ -0,0 +1,215 @@
+      config: {
+        dataSourceEntityType: 'village',
+      },
+    },
+  ];
+
+  const subDistrictAggregationConfig = {
+    type: 'SUM_PER_ORG_GROUP',
+    config: {
+      dataSourceEntityType: 'village',
+      aggregationEntityType: 'sub_district',
+    },
+  };
+
+  const villageAggregationConfig = {
+    type: 'SUM',
+  };
+
+  if (reportCode.includes('Sub_District')) {
+    aggregations.push(subDistrictAggregationConfig);
+  } else if (reportCode.includes('Village')) {
+    aggregations.push(villageAggregationConfig);
+  }
+
+  return {
+    code: reportCode,
+    config: {
+      fetch: {
+        dataElements,
+        aggregations,
+      },
+      transform: [
+        'keyValueByDataElementName',
+        {
+          transform: 'mergeRows',
+          groupBy: 'organisationUnit',
+          using: {
+            period: 'exclude',
+          },
+        },
+        {
+          where: `=(not exists($${dataElements[0]})) or (not exists($most_recent_WS_population))`,
+          transform: 'excludeRows',
+        },
+        {
+          insert: {
+            value: `=min($${dataElements[0]} / $most_recent_WS_population,1)`,
+            organisationUnitCode: '=$organisationUnit',
+          },
+          exclude: '*',
+          transform: 'updateColumns',
+        },
+      ],
+    },
+  };
+};
+
+const PERMISSION_GROUP = 'COVID-19';
+
+const getMapOverlay = (name, reportCode) => ({
+  id: reportCode,
+  report_code: reportCode,
+  legacy: false,
+  name,
+  userGroup: PERMISSION_GROUP,
+  dataElementCode: 'value',
+  isDataRegional: true,
+  measureBuilder: 'useReportServer',
+  measureBuilderConfig: {
+    dataSourceType: 'custom',
+    reportCode,
+  },
+  presentationOptions: {
+    scaleType: 'performance',
+    valueType: 'percentage',
+    displayType: 'shaded-spectrum',
+    scaleBounds: {
+      left: {
+        max: 0,
+        min: 0,
+      },
+      right: {
+        max: 1,
+        min: 1,
+      },
+    },
+    measureLevel: reportCode.includes('Sub_District') ? 'SubDistrict' : 'Village',
+  },
+  countryCodes: '{"WS"}',
+  projectCodes: '{covid_samoa}',
+});
+
+const addMapOverlay = async (db, parentCode, config) => {
+  const { reportCode, dataElements, name, sortOrder } = config;
+  const report = getReport(reportCode, dataElements);
+  const mapOverlay = getMapOverlay(name, reportCode);
+  const permissionGroupId = await nameToId(db, 'permission_group', PERMISSION_GROUP);
+  await insertObject(db, 'report', {
+    id: generateId(),
+    permission_group_id: permissionGroupId,
+    ...report,
+  });
+  await insertObject(db, 'mapOverlay', mapOverlay);
+
+  const mapOverlayGroupId = await codeToId(db, 'map_overlay_group', parentCode);
+  return insertObject(db, 'map_overlay_group_relation', {
+    id: generateId(),
+    map_overlay_group_id: mapOverlayGroupId,
+    child_id: mapOverlay.id,
+    child_type: 'mapOverlay',
+    sort_order: sortOrder,
+  });
+};
+
+const MAP_OVERLAYS = [
```

Comment:
It is up to you, we can make `MAP_OVERLAYS` as an object since it only have one element.
 e.g.: `{reportCode:'', children: []}`